lib/features/common/widgets/user_card.dart

```diff
@@ -88,10 +88,35 @@ class _UserCardState extends State<UserCard>
     );
   }
 
+  Widget favouriteIcon()
+  {
+    return  Container(
+              margin: EdgeInsets.only(right:0),
+              child: Consumer<FavouriteUsersData>(
+                builder: (context,favUsersData,child){
+                  bool isPresent = favUsersData.isPresent(user.id);
+                    return IconButton(
+                      icon: Icon((isPresent)?Icons.favorite_rounded:Icons.favorite_border_rounded,color: Colors.grey,),
+                      onPressed: (){
+                        if(isPresent)
+                        {
+                          favUsersData.removeFromFavourites(user.id);
+                        }
+                        else
+                        {
+                          favUsersData.addToFavourites(user);
+                        }
+                      },
+                    );
+                }
+              ),
+            );
+  }
+
```

Comment:
We can create a Sub-widget here. Instead of creating a method returning a widget. This way  the widget would be more readable and reusable.